infra/stream/InputStream.hpp

```diff
@@ -63,13 +63,53 @@ namespace infra
         StreamErrorPolicy& errorPolicy;
     };
 
+    template<class Parent, class TheReader>
+    class InputStreamWithReader
+        : private detail::StorageHolder<TheReader, InputStreamWithReader<Parent, TheReader>>
+        , public Parent
+    {
+    public:
+        InputStreamWithReader();
+        template<class Arg>
+        explicit InputStreamWithReader(Arg&& arg, std::enable_if_t<!std::is_same_v<InputStreamWithReader, std::remove_cv_t<std::remove_reference_t<Arg>>>, std::nullptr_t> = nullptr);
```

Comment:
I created a wrapper in util/Compatibility.hpp for enable_if_t. Do we still want to provide C++11 compatibility or do we drop that completely?